doc/classes/String.xml

```diff
@@ -181,6 +181,14 @@
 			<argument index="1" name="placeholder" type="String" default="&quot;{_}&quot;" />
 			<description>
 				Formats the string by replacing all occurrences of [code]placeholder[/code] with [code]values[/code].
+				[b]Note:[/b] There must be one and only one [code]_[/code] in the [code]placeholder[/code].
+				[b]Note:[/b] The [code]values[/code] can be a dictionary, an one-dimensional array, or a two-dimensional array. When it's a two-dimensional array, the length of the inner arrays must be [code]2[/code], or an error will be generated.
+				[b]Note:[/b] The function will iterate over the [code]values[/code]. For a dictionary, first, it will replace the [code]_[/code] with the key; Then use the result to find the first matched word; At last, use the value to replace the result (see [method replace]). For an one-dimensional array, like the dictionary, but the index number acts as the key, the element of the array acts as the value. For a two-dimensional array, the first element of the inner array acts as the key, the second one acts as the value.
```

Comment:
> find the first matched word

It uses `replace()` so the function will replace all occurrences of that result.

You can also omit the indices when using an array:

```gdscript
# Prints: User 42 is Godot.
print("User {} is {}.".format([42, "Godot"], "{}"))
```

Reply:
Hmm, but this way it's replaced in order.